src/kernel-install/90-loaderentry.install.in

```diff
@@ -49,6 +49,10 @@ LOADER_ENTRY="$BOOT_ROOT/loader/entries/$ENTRY_TOKEN-$KERNEL_VERSION.conf"
 
 case "$COMMAND" in
     remove)
+        [ -f "/lib/modules/$KERNEL_VERSION/vmlinuz" ] && \
+        [ "$KERNEL_INSTALL_BOOT_ENTRY_TYPE" = "type2" ] && \
+            echo "BOOT_ENTRY_TYPE=type2, not removing loader entries." && \
```

Comment:
This `echo` needs to be guarded by `[ "$KERNEL_INSTALL_VERBOSE" -gt 0 ]` too. (Actually, it's more of a debug stmt, so something like `-ge 2` would be better, but we currently don't implement setting KERNEL_INSTALL_VERBOSE like this, so `-gt 0` like everywhere else is good enough.)